src/librustc_resolve/lib.rs

```diff
@@ -3010,6 +3010,20 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
             let code = DiagnosticId::Error(code.into());
             let mut err = this.session.struct_span_err_with_code(base_span, &base_msg, code);
 
+            // Emit help message for fake-self from other languages like `this`(javascript)
+            let fake_self: Vec<Ident> = ["this", "my"].iter().map(
+                |s| Ident::from_str(*s)
+            ).collect();
+            if fake_self.contains(&item_str)
+                && this.self_value_is_available(path[0].span, span) {
+                err.span_suggestion_with_applicability(
+                    span,
+                    "do you mean",
+                    "self".to_string(),
+                    Applicability::MachineApplicable,
```

Comment:
Hm, is this truly machine applicable in the sense that I can run `cargo fix`, it will apply this, and we can guarantee this fixes the code?

Reply:
I would keep this as `MaybeIncorrect` unless we added a check to see if the binding is available in `self` to begin with (which I don't see in this PR and shouldn't necessarily block it).